tests/robots/test_configuration.py

```diff
@@ -56,22 +73,44 @@ def test_to_data():
     # types=[Joint.PRISMATIC, Joint.REVOLUTE]))
     data = config.to_data()
 
-    assert data['values'] == [8.312, 1.5, 0., 0., 0., 1., 0.8]
+    assert data['joint_values'] == [8.312, 1.5, 0., 0., 0., 1., 0.8]
     assert data['types'] == [Joint.PRISMATIC] + [Joint.REVOLUTE] * 6
 
 
-def test_config_merge():
-    config = Configuration(values=[1, 2, 3], types=[Joint.REVOLUTE]*3, joint_names=['a', 'b', 'c'])
-    other_config = Configuration(values=[3, 2, 0], types=[Joint.REVOLUTE]*3, joint_names=['a', 'b', 'd'])
-    config.merge(other_config)
-    assert config.joint_dict == {'a': 3, 'b': 2, 'c': 3, 'd': 0}
+def test_config_merged():
+    config = Configuration(joint_values=[1, 2, 3], types=[Joint.REVOLUTE]*3, joint_names=['a', 'b', 'c'])
+    other_config = Configuration(joint_values=[3, 2, 0], types=[Joint.REVOLUTE]*3, joint_names=['a', 'b', 'd'])
+    new_config = config.merged(other_config)
+    assert new_config.joint_dict == {'a': 3, 'b': 2, 'c': 3, 'd': 0}
 
 
-def test_joint_trajectory_point_merge():
-    tjp = JointTrajectoryPoint(values=[1, 2, 3], types=[Joint.REVOLUTE]*3, velocities=[4, 5, 6])
+def test_joint_trajectory_point_merged():
+    tjp = JointTrajectoryPoint(joint_values=[1, 2, 3], types=[Joint.REVOLUTE]*3, velocities=[4, 5, 6])
     tjp.joint_names = ['a', 'b', 'c']
-    other_tjp = JointTrajectoryPoint(values=[3, 2, 0], types=[Joint.REVOLUTE]*3, velocities=[0, 5, 0])
+    other_tjp = JointTrajectoryPoint(joint_values=[3, 2, 0], types=[Joint.REVOLUTE]*3, velocities=[0, 5, 0])
     other_tjp.joint_names = ['a', 'b', 'd']
-    tjp.merge(other_tjp)
-    assert tjp.joint_dict == {'a': 3, 'b': 2, 'c': 3, 'd': 0}
-    assert tjp.velocity_dict == {'a': 0, 'b': 5, 'c': 6, 'd': 0}
+    new_tjp = tjp.merged(other_tjp)
+    assert new_tjp.joint_dict == {'a': 3, 'b': 2, 'c': 3, 'd': 0}
+    assert new_tjp.velocity_dict == {'a': 0, 'b': 5, 'c': 6, 'd': 0}
+
+
+def test_dict_behavior():
+    config = Configuration(joint_values=[1, 2, 3], types=[Joint.REVOLUTE]*3, joint_names=['a', 'b', 'c'])
+    assert list(config.items()) == [('a', 1), ('b', 2), ('c', 3)]
+    assert list(config.keys()) == ['a', 'b', 'c']
+    assert list(config.values()) == [1, 2, 3]
+    with pytest.raises(KeyError):
+        _ = config['DNE']
+    assert config['a'] == 1
+    config.joint_values[0] = 4
+    assert config['a'] == 4
+    assert config.get('a') == 4
+    assert config.get('DNE') is None
+    assert config.get('DNE', 5) == 5
+    with pytest.raises(TypeError):
+        config['a'] = 1
+    with pytest.raises(TypeError):
+        config.joint_names.append('d')
+    with pytest.raises(TypeError):
+        config.types[1:1] = range(5)
```

Comment:
i'd suggest to split some of these checks to give them a name, for example, here something like `test_length_immutability` or similar).